cpp/plugins/cucim.kit.cuslide2/src/cuslide/tiff/ifd.cpp

```diff
@@ -0,0 +1,1383 @@
+                            memset(dest_start_ptr + dest_pixel_index, background_value, nbytes_tile_pixel_size_x);
+                        }
+                    }
+                    else
+                    {
+                        cudaError_t cuda_status;
+                        CUDA_ERROR(cudaMemset2D(dest_start_ptr + dest_pixel_index, dest_pixel_step_y, background_value,
+                                                nbytes_tile_pixel_size_x,
+                                                tile_pixel_offset_ey - tile_pixel_offset_sy + 1));
+                    }
+                }
+                } catch (const std::exception& e) {
+                    #ifdef DEBUG
+                    fmt::print("❌ decode_func: Exception caught: {}\n", e.what());
+                    #endif
+                    throw;
+                } catch (...) {
+                    #ifdef DEBUG
+                    fmt::print("❌ decode_func: Unknown exception caught\n");
+                    #endif
+                    throw;
+                }
+            };
+
+            #ifdef DEBUG
+            fmt::print("🔍 read_region_tiles: decode_func lambda created\n");
+            #endif
+
+            // TEMPORARY: Force single-threaded execution to test if decode works
+            bool force_single_threaded = true;
+
+            if (force_single_threaded || !loader || !(*loader))
+            {
+                #ifdef DEBUG
+                fmt::print("🔍 read_region_tiles: Executing decode_func directly (FORCED SINGLE-THREADED TEST)\n");
+                fflush(stdout);
+                #endif
+                decode_func();
+                #ifdef DEBUG
+                fmt::print("🔍 read_region_tiles: decode_func completed successfully!\n");
+                fflush(stdout);
+                #endif
+            }
+            else
+            {
+                #ifdef DEBUG
+                fmt::print("🔍 read_region_tiles: Enqueueing task for tile index={}\n", index);
+                #endif
+                loader->enqueue(std::move(decode_func),
+                                cucim::loader::TileInfo{ location_index, index, tiledata_offset, tiledata_size });
+                #ifdef DEBUG
+                fmt::print("🔍 read_region_tiles: Task enqueued\n");
+                #endif
+            }
+
+            dest_pixel_index_x += nbytes_tile_pixel_size_x;
+        }
+        dest_start_ptr += dest_pixel_step_y * dest_pixel_offset_len_y;
+    }
+
+    return true;
+}
+
+bool IFD::read_region_tiles_boundary(const TIFF* tiff,
+                                     const IFD* ifd,
+                                     const int64_t* location,
+                                     const int64_t location_index,
+                                     const int64_t w,
+                                     const int64_t h,
+                                     void* raster,
+                                     const cucim::io::Device& out_device,
```

Comment:
I wonder when `IFD::read_region_tiles()` and `IFD::read_region_tiles_boundary()` methods are used?
Are they used only when batch processing API are used?